metsim/disaggregate.py

```diff
@@ -7,107 +7,171 @@
-    df_disagg['wind'] = wind(df_daily['wind'])
+                                       solar_geom['tiny_rad_fract'],
+                                       params)
+
+    t_Tmin, t_Tmax = set_min_max_hour(df_disagg['shortwave'], n_days, ts)
+    df_disagg['temp'] = temp(df_daily, df_disagg, t_Tmin, t_Tmax, ts)
+
+    df_disagg['vapor_pressure'] = vapor_pressure(df_daily['vapor_pressure'],
+                                                 df_disagg['temp'],
+                                                 t_Tmin, n_disagg, ts)
+
+    df_disagg['rel_humid'] = relative_humidity(df_disagg['vapor_pressure'],
+                                               df_disagg['temp'])
+
+    df_disagg['longwave'], df_disagg['tskc'] = longwave(df_disagg['temp'],
+                                                        df_disagg['vapor_pressure'],
+                                                        df_daily['tskc'])
+
+    df_disagg['precip'] = precip(df_daily['precip'], ts)
+    df_disagg['wind'] = wind(df_daily['wind'], ts)
+
     return df_disagg.fillna(method='ffill')
 
 
-def set_min_max_hour(disagg_rad, n_days):
+def set_min_max_hour(disagg_rad, n_days, ts):
     """
     TODO
     """
-    ts = float(params['time_step'])
     rad_mask = 1*(disagg_rad > 0)
     diff_mask = np.diff(rad_mask)
-    rise_times = np.where(diff_mask>0)[0] * ts 
+    rise_times = np.where(diff_mask>0)[0] * ts
     set_times = np.where(diff_mask<0)[0] * ts
-    t_Tmax = 0.67 * (set_times - rise_times) + rise_times
-    t_Tmin = rise_times - float(params['time_step'])
-    return t_Tmin, t_Tmax
+    t_t_max = 0.67 * (set_times - rise_times) + rise_times
+    t_t_min = rise_times
+    return t_t_min, t_t_max
 
 
-def temp(df_daily, df_disagg):
+def temp(df_daily, df_disagg, t_t_min, t_t_max, ts):
     """
     TODO
     """
     # Calculate times of min/max temps
-    n_days = len(df_daily['day_of_year'])
-    t_Tmin, t_Tmax = set_min_max_hour(df_disagg['shortwave'], n_days)
     time = np.array(list(next(it) for it in itertools.cycle(
-                [iter(t_Tmin), iter(t_Tmax)])))
+                [iter(t_t_min), iter(t_t_max)])))
     temp = np.array(list(next(it) for it in itertools.cycle(
                 [iter(df_daily['t_min']), iter(df_daily['t_max'])])))
-    
-    try:
-        interp = scipy.interpolate.PchipInterpolator(time, temp, extrapolate=True)
-        temps = interp(float(params['time_step']) * np.arange(0, len(df_disagg.index)))
-    except ValueError:
-        temps = np.full(len(df_disagg.index), np.nan)
-
+    # Account for end points
+    ts_ends = cnst.MIN_PER_HOUR * cnst.HOURS_PER_DAY
+    time = np.append(np.insert(time, 0, time[0:2]-ts_ends), time[-2:]+ts_ends)
+    temp = np.append(np.insert(temp, 0, temp[0:2]), temp[-2:])
+
+    # Interpolate the values
+    interp = scipy.interpolate.PchipInterpolator(time, temp, extrapolate=True)
+    temps = interp(ts * np.arange(0, len(df_disagg.index)))
```

Comment:
The reason this is using scipy is precisely because I don't want to recreate the wheel. `(time, temp)` are a set of coordinates to interpolate over, not Series.  To turn the interpolation step into 1 line we would have to get these into series, which to my knowledge is going to take more than 2 lines.  I just played around with it for a bit and don't really know how to make this happen without adding way more logic than is necessary. Maybe you could figure something out to make this work, but you'll have to rewrite the `set_min_max_hour` function to do this. 

As it stands, this works, and ends up being more efficient than a replacement using pandas exclusively, since we only have to keep 2 points per day instead of all of them.